robot/board.py

```diff
@@ -41,32 +53,53 @@ def _send(self, message, retry=False):
         try:
             self.sock.send(message)
         except (socket.timeout, BrokenPipeError, OSError):
-            if retry:
+            if _is_retry:
```

Comment:
I probably should have asked earlier: what's the purpose of the retries? Is it because we think the connection is flaky or is it a result of using single-message sockets?

If the latter, I'm not convinced that retries are the best way to go. I wonder whether explicitly setting up and tearing down the socket would be better in this case.